operator/pkg/central/extensions/reconcile_admin_password.go

```diff
@@ -116,6 +119,12 @@ func (r *reconcileAdminPasswordExtensionRun) updateStatus(status *platform.Centr
 	if r.infoUpdate == status.Central.AdminPassword.Info {
 		return false
 	}
+
+	if r.autoGenerated || r.hasCredentials {
+		status.Central.AdminPassword.SecretReference = defaultPasswordSecretName
+	} else {
+		status.Central.AdminPassword.SecretReference = r.passwordSecretName
```

Comment:
I'm somewhat worried that the association between the condition and the actions it guards is not very obvious. It *might* be correct in all cases but it seems fragile.
How about in `readPasswordFromReferencedSecret` we instead set `r.passwordSecretName`  to whatever the right secret is (or empty string if none) even if it's the default secret. Then here we would just set `...SecretReference` to either `nil` or that value. Perhaps that would remove the need for the `hasCredentials` field?
Or am I missing some corner case?

Reply:
I really like this idea. Although there is no enough information when `readPasswordFromReferencedSecret` is executed so `passwordSecretName` is changed in downstream functions. The alternative might be to rework password generating/update/reading logic so `readPasswordFromReferencedSecret` will have enough information but I am not sure about implementing such refactoring.
Additional bool var is dropped